db/compaction/compaction_picker.cc

```diff
@@ -39,32 +39,46 @@ bool FindIntraL0Compaction(const std::vector<FileMetaData*>& level_files,
                            size_t min_files_to_compact,
                            uint64_t max_compact_bytes_per_del_file,
                            uint64_t max_compaction_bytes,
-                           CompactionInputFiles* comp_inputs) {
-  size_t compact_bytes = static_cast<size_t>(level_files[0]->fd.file_size);
-  uint64_t compensated_compact_bytes = level_files[0]->compensated_file_size;
+                           CompactionInputFiles* comp_inputs,
+                           SequenceNumber earliest_mem_seqno) {
+
+  // Do not pick ingested file when there is at least one memtable not flushed which of seqno is overlap with the sst.
+  size_t start = 0;
+  while (start < level_files.size() && level_files[start]->being_compacted &&
```

Comment:
Maybe not. If a file is `being_compacted`,  we must skip it.

Reply:
Correct me if I'm wrong: previously, no Inra-L0 compaction is issued if any file scanned is in compaction. Now with the change, we are skipping them instead. I'm not sure whether the new behavior introduces bug or not, but it is really hard to be reason about. In theory, a file can be picked up for L0=>L1 compaction, or another Intro-L0 compaction. Now we are picking older L0 files to an intro-L0 compaction. We are bringing ourselves to a scenario that is really had to reason about. Unless I see a strong case that relaxing the condition can improve performance significantly, my vote would be to keep the logic simple, which is to skip intro-L0 compaction if we see a file in compaction here.

Reply:
Ok, I agree. This is a less risky suggestion.

Reply:
@siying I think it might be a good idea to allow picking *newer* files for intra-L0 compaction while there are L0=>L1 compaction, which makes intra-L0 compaction more useful than before: avoid write stall if there's a slow L0=>L1 compaction.